src/tools/run-make-support/src/rustdoc.rs

```diff
@@ -0,0 +1,97 @@
+use std::env;
+use std::path::Path;
+use std::process::{Command, Output};
+
+use crate::{add_host_rpath_env, handle_failed_output};
+
+pub fn rustdoc() -> RustdocInvocationBuilder {
+    RustdocInvocationBuilder::new()
+}
+
+#[derive(Debug)]
+pub struct RustdocInvocationBuilder {
+    cmd: Command,
+}
+
+impl RustdocInvocationBuilder {
+    fn new() -> Self {
+        let cmd = setup_common_rustdoc_build_cmd();
+        Self { cmd }
+    }
+
+    pub fn arg(&mut self, arg: &str) -> &mut RustdocInvocationBuilder {
+        self.cmd.arg(arg);
+        self
+    }
+
+    pub fn arg_file(&mut self, arg: &Path) -> &mut RustdocInvocationBuilder {
```

Comment:
Should this take some `P: AsRef<Path>`?
```suggestion
    pub fn arg_file<P: AsRef<Path>>(&mut self, arg: P) -> &mut RustdocInvocationBuilder {
```